src/main/java/io/nats/client/impl/NatsJetStream.java

```diff
@@ -256,11 +256,26 @@ NatsJetStreamSubscription createSubscription(String subject, String queueName,
                 createConsumer = false;
                 ConsumerConfiguration lookedUpConfig = lookedUpInfo.getConsumerConfiguration();
 
+                String lookedUp = lookedUpConfig.getDeliverSubject();
```

Comment:
check deliver subject for mismatch to make sure existing consumer is the correct type